core/source_map.rs

```diff
@@ -72,20 +77,69 @@ impl<G: SourceMapGetter> SourceMapper<G> {
       ext_source_maps: Default::default(),
       getter,
       stashed_file_name: Default::default(),
+      maybe_module_map: None,
     }
   }
 
+  pub fn set_module_map(&mut self, module_map: Rc<ModuleMap>) {
+    assert!(self.maybe_module_map.replace(module_map).is_none());
+  }
```

Comment:
See above

Reply:
Actually - we could get `ModuleMap` from the `scope` using the slot. This should simplify things a bit